nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/schema/inference/RecordSource.java

```diff
@@ -18,7 +18,10 @@
 
 import java.io.IOException;
 
-@FunctionalInterface
 public interface RecordSource<T> {
+
+    default void init() throws IOException {
+    }
```

Comment:
Recommend removing this change and implementing the logic in the constructor, similar to the RecordReader.